nevergrad/functions/multiobj/__init__.py

```diff
@@ -0,0 +1,140 @@
+from typing import Callable, List
+from abc import ABC
+import nevergrad
+import numpy as np
+from nevergrad.parametrization import parameter as p
+
+
+class ConstrainedMultiObjective(ABC):
+    def __init__(
+        self,
+        name: str,
+        parametrization: p.Data,
+        objective_funcs: List[Callable[[np.ndarray], float]],
+        constraint_funcs: List[Callable[[np.ndarray], bool]],
+        constraints_as_layer: bool = False,
+    ) -> None:
+        super().__init__()
+        self.name: str = name
+        self.parametrization: p.Data = parametrization
+        self.objective_funcs: List[Callable[[np.ndarray], float]] = objective_funcs
+        self.constraint_funcs: List[Callable[[np.ndarray], bool]] = constraint_funcs
+        self.constraints_as_layer: bool = constraints_as_layer
+        self._register_cheap_constraints()
+
+    def _register_cheap_constraints(self) -> None:
+        for f in self.constraint_funcs:
+            self.parametrization.register_cheap_constraint(f, self.constraints_as_layer)
+
+    def __call__(self, x: np.ndarray) -> List[float]:
+        objectives: List[float] = [f(x) for f in self.objective_funcs]
+        return objectives
```

Comment:
I think this is overly complex, no need to have most of it, only implementing `__call__` and `parametrization` should be enough (eg: defining objective1, objective2 does not bring any more information than just returning the full list of objectives in `__call__`, keeping a list of constraints is not useful since they can be directly applied to the parametrization. In my opinion basically having a function called `function1` is a hint that the function should be just `function` and return something that is indexable)